tools/run_all_tests.py

```diff
@@ -126,6 +129,11 @@ def test_all(
                 break
         else:
             raise ValueError(f"Test to skip '{test_to_skip}' not found in tests.")
+
+    # Filter tests by extension
+    if extension is not None:
+        all_test_paths = [test_path for test_path in all_test_paths if extension in test_path]
```

Comment:
This isn't safe. What is I want to only test `omni.isaac.lab`? With the current check, it will still allow the extensions `omni.isaac.lab_assets` and `omni.isaac.lab_tasks` to be included.

Reply:
Good call! Added a more strict check